deployments/cloudfoundry/bosh/jobs/signalfx-agent/templates/agent.yaml.erb

```diff
@@ -9,11 +9,12 @@ ingestUrl: {"#from": "/var/vcap/jobs/signalfx-agent/config/ingest_url"}
 apiUrl: {"#from": "/var/vcap/jobs/signalfx-agent/config/api_url"}
 cluster: <%= p("signalfx.cluster", "") %>
 
-<% if not p("nozzle.enabled") %>
 globalDimensions:
+<% if not p("nozzle.enabled") %>
   bosh_id: {"#from": "/var/vcap/instance/id", optional: true}
   deployment: {"#from": "/var/vcap/instance/deployment", optional: true}
 <% end %>
+  {"#from": "/var/vcap/jobs/signalfx-agent/config/global-dimensions.yaml", optional: true}
```

Comment:
Should there be `flatten: true` ?

Reply:
Works without it. Any reason to put it?

Reply:
Oh, for maps the key must be `_` for it to be supported.  I'm not sure I understand its use cases so I guess not.

Reply:
For some reason it was not working 08b588a

Reply:
Yes if you want to merge a map into another map, you should do

`_: {"#from": "/file.yaml", optional: true, flatten: true}`

Reply:
Otherwise I would just make the globalDimensions value a YAML map from CF and then loop through it in the ruby template code and render each key/value separately.

Reply:
If the if conditional above this is true then this won't work because it won't be valid yaml anymore.  Test that and see.

Reply:
@keitwb You are correct. Fixed: https://github.com/signalfx/signalfx-agent/pull/1817/commits/d9c7fff37e9d8e2fd534bd8b48a5fd660087642b. I probably did some configuration error when I was testing  https://github.com/signalfx/signalfx-agent/commit/08b588aaeac6651b08ccca4fa19be49d05ea20c5